### PR TITLE
feat: allow grow-only mini-frame resize

### PR DESCRIPTION
**Pull request by kiennq**
This allow `mini-frame` to grow only, similar to setting `resize-mini-window`.
By setting this, it reduced the flickering greatly especially when navigating candidates in `swiper`

### CODE REVIEW

**Comment by muffinmad** [comment]
Did you try to use `fit-frame-to-buffer` and pass current frame height as `min-height` argument?

**Comment by kiennq** [comment]
Ah, that's easier

File: mini-frame.el [hunk 1]
```diff
@@ -154,7 +159,13 @@ Option `resize-mini-frames' is available on Emacs 27 and later."
 (defun mini-frame--resize-mini-frame (frame)
   "Resize FRAME vertically only.
 This function used as value for `resize-mini-frames' variable."
-  (fit-frame-to-buffer frame mini-frame-resize-max-height nil nil nil 'vertically)
```
Comment (muffinmad) [inline mini-frame.el:157]: It can be done in single `fit-frame-to-buffer` call:

```suggestion
(fit-frame-to-buffer frame mini-frame-resize-max-height
                     (when (eq mini-frame-resize 'grow-only)
                       (frame-parameter frame 'height))
                     nil nil 'vertically)
```

Comment (kiennq) [reply mini-frame.el:157]: I purposely separated them so that's easier to see in which condition the each call is done.
Combine them back kind of providing hard to follow logic.
It would be something like
``` elisp
(and (eq mini-frame-resize 'grow-only)
     (active-minibuffer-window)
     (frame-parameter frame 'height))
```

Comment (muffinmad) [reply mini-frame.el:157]: Why need to check `(active-minibuffer-window)`?

Comment (kiennq) [reply mini-frame.el:157]: Without it the `mini-frame` will just keep it height, even after gone to hiding. Which is not nice at all. It should revert to height 1 after exit minibuffer

Comment (muffinmad) [reply mini-frame.el:157]: There are `(height . 1)` in `mini-frame-show-parameters` default value. This parameters apply to mini-frame before each mini-frame display. Do you have some custom `mini-frame-show-parameters` value?

Event without checking `(active-minibuffer-window)` grow-only resize works for me.

Comment (kiennq) [reply mini-frame.el:157]: Oh, I just copied the customization in the package header and overlook the `height . 1` part.
Indeed, the `active-minibuffer-window` is not needed.

Comment (kiennq) [reply mini-frame.el:157]: Removed the check, but I still keep the `(cond)` just for the readability

**Comment by muffinmad** [comment]
Thanks! Merged but with single `fit-frame-to-buffer` call ;)
